packages/core/src/controllers/AccountController.ts

```diff
@@ -239,7 +248,10 @@ export const AccountController = {
         SwapController.setBalances(SwapApiUtil.mapBalancesToSwapTokens(response.balances))
       }
     } catch (error) {
-      SnackController.showError('Failed to fetch token balance')
+      if (chainId && state.tokenFetchFailureChainIds) {
+        state.tokenFetchFailureChainIds.add(chainId)
```

Comment:
Are we not clearing this? What happens if there's a true flake in the request?
I think we should silently retry after some time